Check whether the GMC address is filled at the main Settings page

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

It's based on #936 to prevent code conflicts.

- Check whether the GMC address is filled at the main Settings page.

### Detailed test instructions:

1. Have a saved phone number, and remove address from [Google Merchant Center](https://merchants.google.com/mc/merchantprofile/businessinfo)
    ![image](https://user-images.githubusercontent.com/17420811/128483327-8f50289a-a1db-4324-90ec-10044df3c29c.png)
3. Go to the GLA Settings page, and it should display the no contact information notice
    ![image](https://user-images.githubusercontent.com/17420811/128483552-fc227c8f-dd15-4351-87b9-6ddeb839ca66.png)

### Additional note

💡  If proceeding to the edit contact info page with fulfilled WC store address data, you might found that the "Save detail" button is disabled. This problem was fixed by #937.

### Changelog entry

eason9487 committed:
Check if the GMC address is filled in the ContactInformationPreview

tomalec commented:
:construction: 
I have no address set for MC
`GET /wp-json/wc/gla/mc/contact-information?_locale=user`:
```js
{
	id: 387662089,
	phone_number: null,
	mc_address: null,
	wc_address: {
		street_address: 'Chmielna',
		locality: 'Warszawa',
		region: null,
		postal_code: '01-123',
		country: 'PL',
	},
	is_mc_address_different: true,
}
```
After loading the settings page I got JS error and a blank screen:
```
useStoreAddress.js:62 Uncaught TypeError: Cannot read property 'street_address' of null
    at useStoreAddress (useStoreAddress.js:62)
    at ContactInformationPreview (index.js:35)
    at renderWithHooks (react-dom.js?ver=16.13.1:14938)
    at updateFunctionComponent (react-dom.js?ver=16.13.1:17169)
    at beginWork (react-dom.js?ver=16.13.1:18745)
    at HTMLUnknownElement.callCallback (react-dom.js?ver=16.13.1:182)
    at Object.invokeGuardedCallbackDev (react-dom.js?ver=16.13.1:231)
    at invokeGuardedCallback (react-dom.js?ver=16.13.1:286)
    at beginWork$1 (react-dom.js?ver=16.13.1:23338)
    at performUnitOfWork (react-dom.js?ver=16.13.1:22289)
```

tomalec commented on js/src/hooks/useStoreAddress.js:
```diff
@@ -51,19 +54,18 @@ export default function useStoreAddress() {
-		const city = wcAddress.locality || '';
-		const state = wcAddress.region || '';
-		const postcode = wcAddress.postal_code || '';
+		const streetAddress = storeAddress.street_address || '';
+		const city = storeAddress.locality || '';
+		const state = storeAddress.region || '';
+		const postcode = storeAddress.postal_code || '';
```
[It seems](https://github.com/woocommerce/google-listings-and-ads/pull/938#issuecomment-895196693) we need to cater for `mc_address` `being null` as well.
```suggestion
		const streetAddress = storeAddress?.street_address || '';
		const city = storeAddress?.locality || '';
		const state = storeAddress?.region || '';
		const postcode = storeAddress?.postal_code || '';
```

tomalec commented on js/src/hooks/useStoreAddress.js:
```diff
@@ -51,19 +54,18 @@ export default function useStoreAddress() {
 
 		const [ address, address2 = '' ] = streetAddress.split( '\n' );
-		const country = countryNameDict[ wcAddress.country ];
+		const country = countryNameDict[ storeAddress.country ];
```
cater for `mc_address` `being null` as well.
```suggestion
		const country = countryNameDict[ storeAddress?.country ];
```

ecgan commented:
@tomalec , can you go ahead and make the suggested changes and test it out? If everything works then we can proceed to approve and merge it in.

tomalec commented:
Ok, will do. I'm setting up another Google account to have both scenarios available for testing.

tomalec committed:
Make `useStoreAddress` cater for `mc_address` being `null`

as well.

Otherwise, for merchants that didn't set the store address at all, Settings page would throw and error and render empty.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/938#issuecomment-895196693

tomalec reviewed (approved):
Reviewed the code, and tested it locally (after pushing suggestion):
- Merchant without any address `mc_address: null`
- Merchant with incomplete address 
- Merchant with complete address

Works and LGTM.

:scroll: :nail_care: However, I think it would be nice to have tests for those cases, so we will not forget about them.

ecgan reviewed (approved):
Looking at the code together with @tomalec's changes, LGTM 👍
